node/pkg/common/chainlock.go

```diff
@@ -207,7 +256,115 @@ func UnmarshalOldMessagePublicationWithTxHash(data []byte) (*MessagePublication,
 	return msg, nil
 }
 
-// UnmarshalMessagePublication deserializes a MessagePublication
+// MarshalBinary implements the BinaryMarshaler interface for MessagePublication.
+func (msg *MessagePublication) MarshalBinary() ([]byte, error) {
+	// Marshalled Message Publication layout:
+	//
+	// - TxID length
+	// - TxID
+	// - Timestamp
+	// - Nonce
+	// - Sequence
+	// - ConsistencyLevel
+	// - EmitterChain
+	// - EmitterAddress
+	// - IsReobservation
+	// - Unreliable
+	// - verificationState
+	// - Payload length
+	// - Payload
+
+	const (
+		// TODO: is this big enough?
+		PayloadSizeMax = math.MaxUint16
```

Comment:
What's the reasoning for adding a size cap at all?

Reply:
IMO it's good to always have an explicit upper bound rather than doing unlimited reads. I guess there's some implicit constraint which would be the max of the payload size that any chain could send? I'd rather have one configured though

Reply:
I think if you want to cap the size of the payload, you need to socialize this more widely to make sure it will not impact any integrators.